src/easyscience/Fitting/minimizers/engine_lmfit.py

```diff
@@ -2,36 +2,34 @@
 #  SPDX-License-Identifier: BSD-3-Clause
 #  © 2021-2023 Contributors to the EasyScience project <https://github.com/easyScience/EasyScience
 
-__author__ = "github.com/wardsimon"
-__version__ = "0.1.0"
+__author__ = 'github.com/wardsimon'
+__version__ = '0.1.0'
 
 import inspect
+from typing import Callable
 from typing import List
 from typing import Optional
 
+import numpy as np
 from lmfit import Model as lmModel
-
-# Import lmfit specific objects
 from lmfit import Parameter as lmParameter
 from lmfit import Parameters as lmParameters
 from lmfit.model import ModelResult
 
-from easyscience.Fitting.fitting_template import Callable
-from easyscience.Fitting.fitting_template import FitError
-from easyscience.Fitting.fitting_template import FitResults
-from easyscience.Fitting.fitting_template import FittingTemplate
-from easyscience.Fitting.fitting_template import NameConverter
-from easyscience.Fitting.fitting_template import np
+from .minimizer_base import MinimizerBase
+from .utils import FitError
+from .utils import FitResults
+from .utils import NameConverter
 
 
-class lmfit(FittingTemplate):  # noqa: S101
+class LmFit(MinimizerBase):  # noqa: S101
```

Comment:
I am not sure about CamelCase here.
LM stands for Levenberg-Marquardt - maybe `LMFit` ? This would be similar to other acronym-containing class names. Did we agree on naming?

Reply:
Yes, you are right.  It seems better to have acronyms capitalized in a class name.

Reply:
I just didn't remember if we said anything about acronyms in names, but considering we are dealing with many of those (e.g. QENS, INS) we should have a clear rule

Reply:
In my opinion:
- module names should always be small caps (file tree)
- class names should have CamelCase for the individual words.  If acronym is part of the name it should be ALLCAPS

Reply:
I guess this should be an ADR on the organization level.